src/core/derivation.ts

```diff
@@ -91,6 +91,7 @@ export function shouldCompute(derivation: IDerivation): boolean {
         case IDerivationState.STALE:
             return true
         case IDerivationState.POSSIBLY_STALE: {
+            const prevAllowStateReads = allowStateReadsStart(true)
```

Comment:
Maybe we could add a brief explanation? Something like:
// state propagation can occur outside of action/reactive context
Not sure about the exact phrasing

Reply:
Hm now when I think about it... does the issue persist (without fix) if you modify the test like:
```javascript
const messages = utils.supressConsole(action(() => {
  a.set(1)
}))
```

Reply:
yes